design/ceph/object/store.md

```diff
@@ -119,3 +119,39 @@ Specifying this section also ensures that the pool section in the ceph-object-zo
   zone:
     name: "name"
 ```
+
+### LDAP
+
+The Ceph Object Gateway supports integrating with LDAP for authenticating and creating users, please refer [here](https://docs.ceph.com/docs/master/radosgw/ldap-auth/). This means that the `ceph object store user` is also required to be part of groups in the LDAP server otherwise, authentication will fail. The following settings need to be configured in the RGW server:
+```
+rgw ldap binddn =
+rgw_ldap_secret = /etc/ceph/ldap/bindpass.secret
+rgw ldap uri =
+rgw ldap searchdn =
+rgw ldap dnattr =
+rgw ldap searchfilter =
+rgw_s3_auth_use_ldap = true
+```
+So the CRD for the Ceph Object Store will be modified to include the above changes:
+```yaml
+ldap:
+  uri: ldaps://ldap-server:636
+  binddn: "uid=ceph,cn=users,cn=accounts,dc=example,dc=com"
+  credentialsecret: object-my-store-ldap-creds
+  searchdn: "cn=users,cn=accounts,dc=example,dc=com"
+  dnattr: "uid"
+  searchfilter: "memberof=cn=s3,cn=groups,cn=accounts,dc=example,dc=com"
```

Comment:
Let's make sure there is a separation between configs that are direct 1:1 maps to RGW configs (suggested new `config` section) and configs that aren't (`credentialsecret`). For the credential secret, we will have the most ability to add different credential options in the future if we have a `credential` section.

```suggestion
  credential:
    secret:
      name: <k8s secret name>
  config:
    uri: ldaps://ldap-server:636
    binddn: "uid=ceph,cn=users,cn=accounts,dc=example,dc=com"
    searchdn: "cn=users,cn=accounts,dc=example,dc=com"
    dnattr: "uid"
    searchfilter: "memberof=cn=s3,cn=groups,cn=accounts,dc=example,dc=com"
```

We can easily support users who wish to inject credentials from vault if we allow users to omit the `credential` section. It also allows users to build the creds directly into the container image if they want. Here's how we documented that ability for NFS: https://github.com/BlaineEXE/rook/blob/f907852cac1096e51401247fb34467de79c87cee/Documentation/Storage-Configuration/NFS/nfs-security.md#L129-L132

Reply:
I'm not very enthusiastic about naming config maps "config" but I could live with this.